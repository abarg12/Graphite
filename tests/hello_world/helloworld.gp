int i = 2; 
float x = 3; 
printf(x);