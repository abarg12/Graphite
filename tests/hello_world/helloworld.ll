; ModuleID = 'Graphite'
source_filename = "Graphite"

@a = global i32 2

define void @main() {
entry:
  store i32 2, i32* @a, align 4
  ret void
}
