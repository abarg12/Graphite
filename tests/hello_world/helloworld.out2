; ModuleID = 'Graphite'
source_filename = "Graphite"

@i = global i32 2
@x = global [15 x i8] c"\22howdy Richard\22"
@0 = private unnamed_addr constant [21 x i8] c"\22howdy Richardssss\22\0A\00", align 1

declare i32 @printf(i8*, ...)

define void @main() {
entry:
  store i32 2, i32* @i, align 4
  store [15 x i8] c"\22howdy Richard\22", [15 x i8]* @x, align 1
  %0 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([21 x i8], [21 x i8]* @0, i32 0, i32 0))
  ret void
}
