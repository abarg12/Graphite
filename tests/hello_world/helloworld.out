; ModuleID = 'Graphite'
source_filename = "Graphite"

@i = global i32 2

define void @main() {
entry:
  store i32 2, i32* @i, align 4
  ret void
}
